metadata/metadata.go

```diff
@@ -164,6 +166,23 @@ func ExtractMeta(data []byte, spaceFromCli string, titleFromH1 bool, parents []s
 		meta.Parents = append(parents, meta.Parents...)
 	}
 
+	// deterministically generate a hash from the page's parents, space, and title
+	if titleAppendGeneratedHash {
+		log.Tracef(
+			nil,
+			"appending hash to page title: %s",
+			meta.Title,
+		)
+		path := strings.Join(append(meta.Parents, meta.Space, meta.Title), "/")
+		pathHash := sha256.Sum256([]byte(path))
+		meta.Title = fmt.Sprintf("%s - %x", meta.Title, pathHash[0:4])
```

Comment:
```suggestion
		meta.Title = fmt.Sprintf("%s - %x", meta.Title, pathHash[:7])
```

let's take 7 characters here, just to be safe (in case someone generates a lot of pages)

Reply:
`pathHash` here is an array of bytes being printed as hexadecimal, so each byte is printed as two characters. The first four bytes and the first eight hexadecimal characters are the same, so I could change it to `[:7]` of the hex string if you think that is more clear. 

Either way, I'll add a comment there since it isn't clear what's happening.

Reply:
Ah I didn't see that it's a byte array and hex representation. This looks good (and a comment is a goo idea). Ignore my suggestion then :)

Reply:
I've added a comment for clarity, and included other suggestions.

Thanks for the review!